lib/socket/handlers.js

```diff
@@ -520,6 +523,46 @@ module.exports =  function(socketUtils){
     }, socket.request.sessionId, 'ctrl');
   });
 
+  var updatePresentationSettings = coroutine(function *updatePresentationSettingsGen(socket, evt) {
+    console.log('updatePresentationSettings');
+    var user = socket.decoded_token.user;
+    var userId = user ? user._id : null;
+    var presentation = yield Slideshow.findById(evt.presentationId).exec();
```

Comment:
If the userId doesn't exist you don't need to retrieve the presentation from the database. So make the check first and then retrieve it